sound/soc/sof/intel/hda-dai.c

```diff
@@ -446,6 +446,10 @@ int sdw_hda_dai_hw_params(struct snd_pcm_substream *substream,
 	}
 
 	ops = hda_dai_get_ops(substream, cpu_dai);
+	if (!ops) {
+		dev_err(cpu_dai->dev, "DAI widget ops not set\n");
+		return -EINVAL;
+	}
```

Comment:
this is intentionally not checked as at this point the `hda_dai_get_ops()` will never return NULL.

Reply:
> this is intentionally not checked as at this point the `hda_dai_get_ops()` will never return NULL.

Ok, it was checked by non_hda_dai_hw_params() already. Close the PR now.